src/useCases/submitRegistration.ts

```diff
@@ -0,0 +1,39 @@
+import { IAppContainer } from "../lib/appContainer";
+import { referenceNumber } from "../lib/utils";
+
+export type SubmitRegistrationFn = (
+  submissionId: string
+) => Promise<ISubmitRegistrationResult>;
+
+export interface ISubmitRegistrationResult {
+  beaconRegistered: boolean;
+  confirmationEmailSent: boolean;
+  registrationNumber: string;
+}
+
+export const submitRegistration = ({
+  sendConfirmationEmail,
+  getCachedRegistration,
+  getAccessToken,
+  beaconsApiGateway,
+}: Partial<IAppContainer>): SubmitRegistrationFn => async (submissionId) => {
+  const registration = await getCachedRegistration(submissionId);
+  const accessToken = await getAccessToken();
+
+  const beaconRegistered = await beaconsApiGateway.sendRegistration(
```

Comment:
The reference number needs to be set before it is sent to the API, otherwise this will not be persisted in the API and will not be sent in the confirmation email